concepts/0094-cross-domain-messaging/README.md

```diff
@@ -0,0 +1,302 @@
+- From the DIDDoc, a service entry for each Receiver agent that includes:
+  - a public DID or physical endpoint and public key for the endpoint agent.
+  - An optional order list of public keys (`RoutingKeys`) of mediators used by the Receiver in delivering the message.
+
+The DIDDoc will have a public key entry for each additional Agent message Receiver and each mediator.
+
+In many cases, the entry for the endpoint agent should be a public DID, as it will likely be operated by an agency (for example, https://agents-r-us.com) rather than by the Receiver entity (for example, a person). By making that a public DID in that case, the agency can rotate its public key(s) for receiving messages in a single operation, rather than having to notify each identity owner and in turn having them update the public key in every pairwise DID that uses that endpoint.
+
+#### Data Not Exposed
+
+Given the sequence specified above, the following data is **NOT** exposed to the Sender's agents:
+
+- The relays within the Receiver's domain
+- Receiver agents the identity chooses not to include in a given DIDDoc
+- The physical endpoints of agents within the Receiver's domain (other than the Domain Endpoint)
+  - The physical endpoints and, as required, associated public keys, are shared as needed by the agents within the Receiver's domain.
+
+## Message Types
+
+The following Message Types are defined in this RFC.
+
+### Core:Routing:1.0:Forward
+
+The core message type "forward", version 1.0 of the "routing" family is defined in this RFC. An example of the message is the following:
+
+```json
+{
+  "@type" : "did:sov:BzCbsNYhMrjHiqZDTUASHg;spec/routing/1.0/forward",
+  "to"   : "did:sov:1234abcd#4",
+  "msg"  : { json object from <pack(AgentMessage,valueOf(did:sov:1234abcd#4), privKey(A.did@A:B#1))> }
+}
+```
+
+The `to` field is required and takes one of two forms:
+
+- A DID without a key reference (e.g. `did:sov:1234abcd`)
+- A DID with a key reference (e.g. `did:sov:1234abcd#4`)
+
+The first form is used when sending forward messages across one or more agents that do not need to know the details of a domain. The Receiver of the message is the designated `Routing Agent` in the Receiver Domain, as it controls the key used to decrypt messages sent to the domain, but not to a specific Agent.
+
+The second form is used when the precise key (and hence, the Agent controlling that key) is used to encrypt the Agent Message placed in the `msg` field.
+
+The `msg` field calls the Indy-SDK `pack()` function to encrypt the Agent Message to be forwarded. The Sender calls the `pack()` with the suitable arguments to AnonCrypt or AuthCrypt the message. The `pack()` and `unpack()` functions are described in more detail in the [Encryption Envelope](../../features/0019-encryption-envelope/README.md) RFC.
+
+## Reference
+[reference]: #reference
+
+See the other RFCs referenced in this document:
+
+),  and . Those RFCs should be considered together in understanding  messaging.
```

Comment:
@swcurran typo? Maybe the line is unnecessary?

Reply:
Yes - typo.  Removed.